### DISCUSSION TITLE
JAMF Configuration Profile JSON Schema

### DISCUSSION THREAD

**Discussion by demck85** [jamf/Setup-Manager#4] [Ideas]
would be nice to have a JAMF JSON Schema for the config profile

**Comment by scriptingosx** [comment]
We are working on it. There are some challenges with the localized keys that don't represent well in the Jamf Pro custom interface

**Comment by maumeerally** [comment]
I've built one, it's a work-in progress but it works (see attached file).
[com.jamf.setupmanager.json](https://github.com/user-attachments/files/16265742/com.jamf.setupmanager.json)

**Comment by maumeerally** [comment]
Whilst the JSON Schema I've made works fine, I do have an issue uploading it into Jamf Pro and have raised support ticket CS1195808 for it.

In the meantime I use the JSON Schema in a config profile with a different preference domain (com.myorg.setupmanager) and then take the XML Plist that has been generated by the profile and re-upload it into a separate new profile via the plist upload option with the correct preference domain of com.jamf.setupmanager

**Comment by duggan1972** [comment]
I've noticed there are only 2 options when Jamf Setup Manager ends: Shut Down and Continue. It would be great to have a Restart or forcedRestart as well. Are there plans to expand on those? I've been testing this as a replacement for DEPNotify. I've configured Setup Your Mac as well, however Jamf Setup Manager may be a better fit for our organization.

**Reply by scriptingosx** [reply]
we have gotten this request and are looking into it.

Is this for User Initiated enrollment in particular? because when you are running after automated enrollment, I would recommend using "shutdown." That way, Setup Assistant will start over, and, for example, prompt for Wifi connection again when the previously setup network is not available...

**Reply by duggan1972** [reply]
We use ABM, and our enrollments always have a PreStage. However, we have a complicated setup workflow that requires the Mac to be Restarted. Of course, a Shutdown solves this as well, it's just not as smooth as a Forced Restart. Even a Forced Logout would work. We want the User to not have an option after Jamf Setup Manager runs. They either need to Restart/Log Out/Shut Down. For now, I believe I can make the Shut Down button the only Button visible

**Reply by scriptingosx** [reply]
you can set `finalAction` to `shutdown` and `finalCountdown` to something really short, even a value of `1`

**Reply by scriptingosx** [reply]
I would like to understand this 'complicated setup workflow that requires the Mac to be restarted.' if you want to elaborate on that, ping me on Mac Admins Slack, same username

**Comment by scriptingosx** [comment]
see https://github.com/Jamf-Concepts/Setup-Manager/discussions/21
